airbyte-integrations/connectors/source-mysql/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceOperations.kt

```diff
@@ -215,9 +215,12 @@ class MysqlSourceOperations :
                 // of table on very large tables ( > 100s million of rows)
                 val greatestRate: String = 0.00005.toString()
                 // Quick approximation to "select count(*) from table" which doesn't require
-                // full table scan.
+                // full table scan. However, note this could give delayed summary info about a table
+                // and thus a new table could be treated as empty despite we recently added rows.
+                // To prevent that from happening and resulted for skipping the table altogether,
+                // the minimum count is set to 10.
                 val quickCount =
-                    "SELECT table_rows FROM information_schema.tables WHERE table_schema = '$namespace' AND table_name = '$name'"
+                    "SELECT GREATEST(10, table_rows) FROM information_schema.tables WHERE table_schema = '$namespace' AND table_name = '$name'"
```

Comment:
I'm not convinced this is working. I have:

```sql
(SELECT GREATEST(10, table_rows) FROM information_schema.tables WHERE table_schema = 'sghost05_wp789' AND table_name = 'view_orders')
```

that returns null on my end. MySQL v8.0.39-30.

We used to have correct results with 

<img width="820" alt="SCR-20241210-nglt" src="https://github.com/user-attachments/assets/a4892fc2-46b7-4d13-9754-080ead658717">

https://cloud.airbyte.com/workspaces/9e090881-de2e-48c7-b268-2e86ad2525d9/connections/b46a92b8-8aa3-44e9-b17a-26067781a1cc/timeline

and now since https://cloud.airbyte.com/workspaces/9e090881-de2e-48c7-b268-2e86ad2525d9/connections/b46a92b8-8aa3-44e9-b17a-26067781a1cc/timeline

<img width="718" alt="SCR-20241210-ngqb" src="https://github.com/user-attachments/assets/e8a9ccd6-9c30-41f6-9f6a-eb12e0777438">

it's empty.

Reply:
Moved to a dedicated issue #49804.